lib/dsc-lib/src/functions/json.rs

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crate::DscError;
+use crate::configure::context::Context;
+use crate::functions::{FunctionArgKind, Function, FunctionCategory, FunctionMetadata};
+use rust_i18n::t;
+use serde_json::Value;
+
+#[derive(Debug, Default)]
+pub struct Json {}
+
+impl Function for Json {
+    fn get_metadata(&self) -> FunctionMetadata {
+        FunctionMetadata {
+            name: "json".to_string(),
+            description: t!("functions.json.description").to_string(),
+            category: vec![FunctionCategory::Object],
+            min_args: 1,
+            max_args: 1,
+            accepted_arg_ordered_types: vec![
+                vec![FunctionArgKind::String],
+            ],
+            remaining_arg_accepted_types: None,
+            return_types: vec![FunctionArgKind::Object, FunctionArgKind::Array, FunctionArgKind::String, FunctionArgKind::Number, FunctionArgKind::Boolean],
+        }
+    }
+
+    fn invoke(&self, args: &[Value], _context: &Context) -> Result<Value, DscError> {
+        let Some(json_str) = args[0].as_str() else {
+            return Err(DscError::Parser(t!("functions.json.notString").to_string()));
+        };
```

Comment:
Since this function ONLY accepts a string and the input param type is pre-validated before calling this implementation, that error message should never be actually used.  I think it's simpler to just use:

```rust
let json_str = args[0].as_str().unwrap();
```

Since validation already happened, this should never panic.